### PR TITLE
Simplify deployment

### PR DESCRIPTION
**Pull request by AP-Hunt**
What
---

Route services are kinda complicated, involving a couple of moving parts. We could make this a bit easier for users of this route service by automating some of that.

How to review
---
1. Code review
2. Deploy an app you want to protect with the basic auth route service
3. Run the deploy script, targeting the app you want to protect.
4. Check it runs successfully
5. See if your app has been protected

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -11,3 +11,16 @@ can be set with the `cf set-env` command.
...
+## Deployment script
+You can deploy this route service to GOV.UK PaaS using the deployment in `deploy.sh`. It will not work outside of GOV.UK PaaS, because of some assumptions made about domain names.
+
+To run the script, see the example below:
+```shell script
+AUTH_USERNAME=username \
+AUTH_PASSWORD=password \
+ROUTE_SERVICE_APP_NAME=name_of_this_app \
+ROUTE_SERVICE_NAME=name_to_give_the_route_service \
+PROTECETD_APP_NAME=name_of_the_app_to_protect_with_basic_auth \
```
Comment (schmie) [inline README.md:24]: typo: PROTECTED_APP_NAME=name_of_the_app_to_protect_with_basic_auth \

File: deploy.sh [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  echo "Must set ROUTE_SERVICE_NAME environment variable"
+  exit 1
+fi
+
+if [ -z ${PROTECTED_APP_NAME+set} ]; then
+  echo "Must set PROTECTED_APP_NAME environment variable"
+  exit 1
+fi
+
+API_ENDPOINT="$(cf target | head -n1 | awk '{ print $3 }')"
```
Comment (schmie) [inline deploy.sh:34]: API_ENDPOINT is unused and I had to remove this line - did you want to evaluate it?

**Review by schmie — CHANGES_REQUESTED** [review]
Two minor things to resolve, but then it worked like a charm :D 
https://static-app.schmie.dev.cloudpipelineapps.digital/ was never more secure <3

File: deploy.sh (hunk 2 above)
Comment (AP-Hunt) [reply deploy.sh:34]: No, I think it's leftover from another approach I took to working out the apps domain. Will remove.

File: README.md (hunk 1 above)
Comment (AP-Hunt) [reply README.md]: Good catch

**Review by schmie — APPROVED** [review]
:+1:

**Pull request merged**